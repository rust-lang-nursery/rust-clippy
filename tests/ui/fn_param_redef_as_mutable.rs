#![warn(clippy::fn_param_redef_as_mutable)]

fn foobar(a: Vec<bool>) {
    let mut c = a;
}

fn main() {
}
