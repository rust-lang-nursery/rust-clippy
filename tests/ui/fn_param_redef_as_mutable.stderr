error: a parameter was redefined as mutable, can be removed
  --> $DIR/fn_param_redef_as_mutable.rs:3:1
   |
LL | / fn foobar(a: Vec<bool>) {
LL | |     let mut c = a;
LL | | }
   | |_^
   |
   = note: `-D clippy::fn-param-redef-as-mutable` implied by `-D warnings`
help: consider making this param `mut`
  --> $DIR/fn_param_redef_as_mutable.rs:3:11
   |
LL | fn foobar(a: Vec<bool>) {
   |           ^^^^^^^^^^^^^
help: consider removing this local variable
  --> $DIR/fn_param_redef_as_mutable.rs:4:5
   |
LL |     let mut c = a;
   |     ^^^^^^^^^^^^^^

error: aborting due to previous error

